test/jit/test_symbolic_shape_analysis.py

```diff
@@ -277,7 +277,7 @@ def make_arg(shape, low=None, high=None):
             nn_inps = (
                 (make_arg((40,), 0, 9), torch.nn.Embedding(20, embedding_dim=64, max_norm=1.0)),
                 (make_arg((2, 4), 0, 9), torch.nn.Embedding(10, 20, sparse=True)),
-                (make_arg(()), torch.nn.Embedding(0, 0, sparse=True)),
+                (make_arg((0,)), torch.nn.Embedding(0, 0, sparse=True)),
```

Comment:
What's going on here?

Reply:
`make_arg(())` will create a scalar tensor instead of empty tensor, `nn.Embedding` calls `index_select` so with the pr this test will fail. `make_arg((0,))` creates an empty tensor instead.

Reply:
I'm not sure if this is right or not, but it does seem to me that there is a BC breaking change here (which hopefully doesn't matter). It would be nice to have a more detailed argument why the original logic here was wrong.

Reply:
yeah, the change in `index_select` is BC breaking in the same sense. Out of bound index access of empty tensor/embedding table will error out instead of returning a random tensor. I think 1) one should not rely on these random tensors. 2) this behavior is consistent with non-empty source tensor case of index_select, where we check index bounds.

Reply:
also, index_select_cuda checks out of bound index for empty source tensor.

Reply:
I went ahead and Marked this PR as bc breaking